crates/ethcore/sync/src/chain/propagator.rs

```diff
@@ -230,42 +248,53 @@ impl SyncPropagator {
             }
 
             // Construct RLP
-            let (packet, to_send) = {
```

Comment:
here, we overwrote `to_send` with transactions that we intend to send. and we use it again when we are setting `peer_info.last_sent_transactions`. In the new commit, we will always set all `to_send` not just transactions that we intend to send to peer.

Reply:
`stats` was not main problem, but setting `peer_info.last_sent_transactions` wrongly